sale_team_price_policy/models/price_policy_mixin.py

```diff
@@ -0,0 +1,117 @@
+        elif section.price_policy == \
+                'partner_pricelist_if_exists':
+            if partner.property_product_pricelist:
+                final['pricelist_id'] = partner.property_product_pricelist.id
+            else:
+                final['pricelist_id'] = section.pricelist_id.id
+        elif section.price_policy == 'partner_pricelist':
+            final['pricelist_id'] = partner.property_product_pricelist.id
+        return final
+
+    @api.model
+    def create(self, vals):
+        if vals.get('section_id'):
+            partner = self.env['res.partner'].browse(vals.get('partner_id'))
+            section = self.env['crm.case.section'].browse(vals['section_id'])
+            vals.update(
+                self._synchro_fields(section=section, partner=partner))
+        return super(PricePolicyMixin, self).create(vals)
+
+    # @api.multi
+    # # @api.onchange('pricelist_id')
+    # def onchange_pricelist_id(self, pricelist_id):
+    #     import pdb; pdb.set_trace()
+    #     return super(SaleOrder, self).onchange_pricelist_id(pricelist_id)
+    #     for record in self:
+    #         import pdb; pdb.set_trace()
+    #         if record.pricelist_id and record.state == 'draft':
+    #             for line in record.record_line:
+    #                 if not line.record_line_ids:
+    #                     line.update_from_pricelist()
+
+    # @api.multi
+    # def onchange(self, values, field_name, field_onchange):
+    #     # To avoid hard inheriting on partner_id we always
+    #     # play the section_id
+    #     print "###############field_name############"
+    #     print field_name
+    #     if isinstance(field_name, list) and 'section_id' in field_name:
+    #         field_name.remove('section_id')
+    #         field_name.append('section_id')
+    #     # If we call an onchange on the partner_id we also call the onchange
+    #     # on the section_id for the compatibility
+    #     if field_name == 'partner_id':
+    #         field_name = ['partner_id', 'section_id']
+    #     if field_name == 'pricelist_id':
+    #         field_name = ['pricelist_id', 'section_id']
+    #     return super(SaleOrder, self).onchange(
+    #         values, field_name, field_onchange)
+
+    @api.model
+    def _check_price_policy(self):
+        """
+        Check if price policy of record team are fulfilled.
+        We check only for customer invoice (not supplier invoice nor refunds)
+        """
+        self.ensure_one()
+        if self.section_id.price_policy == 'contract_pricelist' and \
+                self.pricelist_id != self.section_id.pricelist_id:
+            raise UserError(
+                HELP_PRICELIST + u" market price " + HELP_POLICY)
+        if self.section_id.price_policy == 'partner_pricelist_if_exists':
+            if self.partner_id.property_product_pricelist:
+                if self.partner_id.property_product_pricelist \
+                        != self.pricelist_id:
```

Comment:
Pas plus simple de mettre un seul IF:
if self.partner_id.property_product_pricelist and self.partner_id.property_product_pricelist != self.pricelist_id:

Reply:
Bizarrement je trouve plus long mais plus clair car en mettant une condition seulement y a pas d'ambiguité : on sait ce que le elif représente (booelen) meme quand on est fatigué.
Je me méfie des conditions  
```
if: AND/OR
elif:
```
Je trouve tres dur (long) dans quel cas on passe dans le elif. Au moins il faut le relire pluiseurs fois, c qui est génant

Du coup je me dis que sur ce genre d'exemple la concision pourrait être notre ennemi